org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/DatabaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package org.eclipse.scanning.api.database;
+
+/**
+ * 
+ * This class is a database 
+ * 
+ * @author Matthew Gerring
+ *
+ * @param <T>
+ */
+@FunctionalInterface
+public interface DatabaseOperation<T> {
+	public Id operate(T bean) throws Exception;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)